content_scripts/vimium_frontend.coffee

```diff
@@ -96,6 +96,30 @@ settings =
 #
 frameId = Math.floor(Math.random()*999999999)
 
+# If an input grabs the focus before the user has interacted with the page, then grab it back.
+class GrabBackFocus extends Mode
+  constructor: ->
+    super
+      name: "grab-back-focus"
+      keydown: => @alwaysContinueBubbling => @exit()
+
+    @push
+      _name: "grab-back-focus-mousedown"
+      mousedown: => @alwaysContinueBubbling => @exit()
+
+    chrome.storage.sync.get "grabBackfocus", (items) =>
```

Comment:
If those are things we need the settings library to do, we should do them there, rather than introducing implementation details throughout the code that we'll forget to re-integrate later.

I agree that the frontend should use `chrome.storage` directly; I started work on using the backend settings library for the frontend too in #1290, and all those things (except no caching) come for free.